assets/js/datagovsg-search.js

```diff
@@ -201,25 +247,24 @@ function changePage(curr, index) {
     datagovsgOffset += 100;
     databaseSearch(searchTerm, index, searchField);
   }
+  // Always also look 5 pages ahead and behind
+  const forwardIndex = index + 5
+  const backwardIndex = index - 5
+  if (shouldCallAPI(forwardIndex)) {
+    databaseSearch(searchTerm, forwardIndex);
+  }
+  if (shouldCallAPI(backwardIndex)) {
+    databaseSearch(searchTerm, backwardIndex);
+  }
 
   changePageUtil(curr, index);
   displayTable(pageResults[currentPageIndex], fieldArray);
 }
 
-// Evaluates to true if we should call the datagovsg API for the 
-// next 100 rows of data
+// Evaluates to true if we should call the datagovsg API
 function shouldCallAPI(index) {
-  // Checks to make sure that the last digit of the page number is greater than 5.
-  // i.e. we should call the API for the next 100 rows if the user is currently at 
-  // page number 26.
-  // Note: the index starts from 0, so a page index of 14 corresponds to a page number of 15.
-  if (index % 10 < 4) return false;
-
-  // Makes sure that there is more data to be retrieved from the API.
-  if (datagovsgOffset + 100 > datagovsgTotal) return false;
-
-  // Makes sure that we haven't already called the API for the next 100 rows.
-  if (index * RESULTS_PER_PAGE < datagovsgOffset) return false;
-
-  return true;
+  if (pageResults[index] !== null) return false
```

Comment:
Nit: is it easy to disable the buttons instead? else we have a CTA that does nothing with no user feedback

Reply:
Sense check here, when would the condition at line 266 be true but the condition at line 268 be false ah?

Reply:
Buttons are hidden completely when not in range! So there shouldn't be a CTA with no feedback

pageResults actually acts as a check for whether we've queried the data before - pageResults[index] might be not populated because the index requested is not in range, or it might just not have been searched before